src/api/middleware/validators/offer.js

```diff
@@ -135,6 +138,232 @@ const create = useExpressValidators([
+        if (jobMinDuration >= offer.jobMaxDuration.toString() &&
+            !jobMaxDuration) {
+
+            // end date is earlier than publish date, error!
+            throw new Error(ValidationReasons.MUST_BE_BEFORE("jobMaxDuration"));
+        }
+    } catch (err) {
+        throw new Error(err);
+    }
+    return true;
+};
+
+const jobMaxDurationEditable = async (jobMaxDuration, { req }) => {
+    try {
+        const offer = await (new OfferService()).getOfferById(req.params.offerId, req.user);
+
+        const { jobMinDuration: jobMinDurationCandidate } = req.body;
+
+        const jobMinDuration = jobMinDurationCandidate || offer.jobMinDuration.toString();
+        // If the new publishEndDate is after the new publishDate, the verification will be done in publishEndDate
+        if (jobMinDuration >= jobMaxDuration) {
+
+            // end date is earlier than publish date, error!
+            throw new Error(ValidationReasons.MUST_BE_AFTER("jobMinDuration"));
+        }
+    } catch (err) {
+        throw new Error(err);
+    }
+    return true;
+};
+
+const publishDateEditable = async (publishDateRaw, { req }) => {
+    try {
+        const offer = await (new OfferService()).getOfferById(req.params.offerId, req.user);
+        // const publishDateCandidate = publishDateRaw instanceof Date ? publishDateRaw.toISOString() : publishDateRaw;
+        const { publishEndDate: publishEndDateCandidate } = req.body;
+
+        // If the new publishEndDate is after the new publishDate, the verification will be done in publishEndDate
+        if (publishDateRaw >= offer.publishEndDate.toISOString() &&
+                !publishEndDateCandidate) {
+
+            // end date is earlier than publish date, error!
+            throw new Error(ValidationReasons.MUST_BE_BEFORE("publishEndDate"));
+        }
+
+    } catch (_e) {
+        // Also catches any fail to the DB
+        throw new Error(_e);
+    }
+    return true;
+};
+
+const publishEndDateEditable = async (publishEndDateCandidate, { req }) => {
+    try {
+        // Default values and also handling if it is string or date object
+        const offer = await (new OfferService()).getOfferById(req.params.offerId, req.user);
+        const { publishDate: publishDateCandidate } = req.body;
+
+        let publishDate;
+
+        // Verifies if it's possible to convert the date
+        // If not, the validator will stop running without an error message because it was already thrown in publishDate validator
+        if (publishDateCandidate) {
+            try {
+                publishDate = (new Date(Date.parse(publishDateCandidate))).toISOString();
+            } catch (_e) {
+                return false;
+            }
+        }
+        publishDate = publishDate || offer.publishDate.toISOString();
+
+        // If the new publishEndDate is after the new publishDate, this verification is already done in publishDate
```

Comment:
Forgot about this, I'll verify this aswell

Reply:
Currently implementing this, but I think instead of a throw I should return a custom response, I don't really like the look it will have. Any other suggestion?
`   { "error_code": 1,
    "errors": [
        {
            "value": "2022-03-01",
            "msg": "cant-exceed:6-got:12.00",
            "param": "publishEndDate",
            "location": "body"
        }
    ]
}`

Reply:
Or maybe I should just return the ammount of months passed, like here https://github.com/NIAEFEUP/nijobs-be/pull/110#discussion_r567444139

Reply:
Just to clarify: there's nothing needed here, right? Regarding this PR, at least